### PR TITLE
Websockets

### PR DESCRIPTION
**Pull request by R4PaSs**
Fixed websockets with the new and improved sockets.

While I'm at it, changed a bit the way it works, the old one was a bit (a lot ?) quick and dirty, so here's that.

### CODE REVIEW

**Comment by privat** [comment]
une idée pour tester les régressions pour éviter qu'on le rebrise sans s'en rendre compte ? Est-ce qu'on pourrait utiliser les testing units http://nitlanguage.org/tools/nitunit/

File: lib/websocket/websocket.nit [hunk 1]
```diff
@@ -39,39 +36,51 @@ class WebSocket
...
 		assert not listener.closed
 
 		var client = listener.accept
 		assert client != null
-		self.client = client
 
+		return new WebsocketConnection(listener.port, "", client)
+	end
+
+	# Disconnects the client if one is connected
```
Comment (Morriar) [inline lib/websocket/websocket.nit]: Les deux lignes pourraient être dans le synopsys sans pour autant dépasser les 80 chars

**Comment by Morriar** [comment]
+1

**Comment by Morriar** [comment]
Jenkins dit que tu as réussi à corriger les 2900 warnings restant, congratz!

Ça ou alors t'as encore pété le bootstrap :)

**Comment by R4PaSs** [comment]
Fixed the documentation issue reported by @Morriar

**Comment by R4PaSs** [comment]
Also, updated more than websockets, this might need a new review

**Comment by Morriar** [comment]
Please document your rerolls.

**Comment by R4PaSs** [comment]
Sure, sorry !

There used to be a new bug introduced in #932 so that when using the interpreter in a mode that redefines the streams from `Sys`, a direct wrapper to standard file descriptors (0, 1 and 2) were used.

Thing is, since we can override them via the `--socket` and `--websocket` options in `niti`, the output from the interpreted program happened locally, regardless of how the inputs and outputs were forwarded.

So, a simple workaround is implemented to ensure the good behaviour here.

And, since the model of websockets changes, an update was needed in the debugger to support it !

Now it seems you're out of excuses, re-review that PR :)

**Comment by Morriar** [comment]
Reroll comment was too long, didn't read :p

**Comment by Morriar** [comment]
+1

**Comment by R4PaSs** [comment]
Reroll, the FIXME-statement disappeared, and String.substring was producing Strings with a negative length in some cases, this should not happen anymore.

**Comment by privat** [comment]
+1

**Pull request merged**